### ISSUE TITLE
SDK requires `payable` for `init` function

### ISSUE THREAD

**Issue by marc0olo** [aeternity/aepp-sdk-js#1436]
seems like we introduced a bad behavior 😞 

in aepp-sophia-examples we provide AE to the `init` function while deploying a contract and this used to work with the previous version.

now with v11.0.0 the SDK complains:
```
NotPayableFunctionError: You try to pay "You try to pay "100000" to function "init" which is not payable. Only payable function can accept coins" to function "undefined" which is not payable. Only payable function can accept tokens
```

independently of that this error message seems to be strange, too.

when I add (for test-purposes) the `payable` declaration to the `init` function the compiler complains as follows:
```
Error: aci error:
type_error:14:3: The init function (at line 14, column 3) cannot be payable.
You don't need the 'payable' annotation to be able to attach
funds to the create contract transaction.
```

**Comment by subhod-i** [comment]
@marc0olo  I see the same compiler complaint in AeStudio too.

**Comment by marc0olo** [comment]
did we change sth. in these checks? it seems like it should be allowed to call `init` with amount > 0 from SDK side

**Comment by davidyuk** [comment]
fixed in #1462
